### ISSUE TITLE
Expose Terraform ResourceId to assert it

### ISSUE THREAD

**Issue by aabouzaid** [stelligent/config-lint#160]
Hello,
Thanks for that great project, it's really helpful.

## What the problem I want to solve
I want to lint Terraform resource id, however only resource arguments are exposed.
For example:
```
resource "aws_instance" "web" {
  ami           = "ami-a1b2c3d4"
  instance_type = "t2.micro"
}
```

I want to apply som opreations on resource name (which is called `web` in this case) (e.g make sure the name is not longer than 20 characters).

I did check the code, and I don't see ResourceId is exposed anywhere.

If `config-lint` doesn't support this yet, could it be added as a feature request.

Best.

**Comment by milldr** [comment]
We're glad at you're making use of our project!

For now, config-lint can only lint against static resources. So resources fields such as ID or ARN that are created dynamically can't be exposed to test

**Comment by aabouzaid** [comment]
@milldr Thanks for your reply Dan.
But the name value is static not dynamic.
Here I don't refer to resource ID that's generated in Terraform, but the actual resource name (the name is the string after the resource type).

It's called `ResourceID` in `config-lint` violation output:
```
[
  {
    "AssertionMessage": "egress should be present",
    "Category": "resource",
    "CreatedAt": "2020-03-23T17:04:11Z",
    "Filename": "/mnt/test.tf",
    "LineNumber": 13,
    "ResourceID": "egress",
    "ResourceType": "aws_security_group",
    "RuleID": "SG_MISSING_EGRESS",
    "RuleMessage": "Security group should specify egress rules",
    "Status": "WARNING"
  }
]
```

**Comment by milldr** [comment]
Ah I understand now. This is also not supported at the moment, since rules can only evaluate resource properties. However that shouldn't be as major of a feature request as adding something dynamic since we already have that value. So I'll keep this open as a feature request

**Comment by aabouzaid** [comment]
Fixed by #164
